swagger.json

```diff
@@ -5200,64 +5200,115 @@
+                    "description": "ID of the Samsara dispatch job that the document is submitted for.",
                     "example": 773
                 },
-                "vehicleId": {
-                    "type": "integer",
-                    "format": "int64",
-                    "description": "VehicleID of the driver at document creation.",
-                    "example": 222
-                },
                 "notes": {
                     "type": "string",
                     "description": "Notes submitted with this document.",
                     "example": "Fueled up before delivery."
-                },
-                "fields": {
-                    "description": "The fields associated with this document.",
-                    "type": "array",
-                    "items": {
-                        "$ref": "#/definitions/DocumentFields"
-                    }
                 }
             }
         },
+        "Document": {
+            "allOf": [
+                {
+                    "type": "object",
+                    "required": [
+                        "driverId",
+                        "driverCreatedAtMs",
+                        "documentType",
+                        "fields"
+                    ],
+                    "properties": {
+                        "driverId": {
+                            "type": "integer",
+                            "format": "int64",
+                            "description": "ID of the driver that submitted this document.",
+                            "example": 555
+                        },
+                        "driverCreatedAtMs": {
+                            "type": "integer",
+                            "format": "int64",
+                            "description": "The time in Unix epoch milliseconds that the document was created by the driver.",
+                            "example": 1462881998034
+                        },
+                        "documentType": {
+                            "type": "string",
+                            "description": "Descriptive name of this type of document.",
+                            "example": "Fuel Receipt"
+                        },
+                        "vehicleId": {
+                            "type": "integer",
+                            "format": "int64",
+                            "description": "VehicleID of the driver at document creation.",
+                            "example": 222
+                        },
+                        "fields": {
+                            "description": "The fields associated with this document.",
+                            "type": "array",
+                            "items": {
+                                "$ref": "#/definitions/DocumentFields"
+                            }
+                        }
+                    }
+                },
+                { "$ref": "#/definitions/DocumentBase"}
```

Comment:
question: Why do we use the `*Base` pattern here (`DocumentBase`), but not for `DocumentField`?

Reply:
Everything in `DocumentFieldCreate` is also in `DocumentField`, but there are some fields in `Document` that aren't in `DocumentCreate` and vice versa, so `DocumentBase` is the things common to both of them.